util/xxhash3/internal/xxh3_raw/xxh3_raw.go

```diff
@@ -0,0 +1,441 @@
+		prime32_3, prime64_1, prime64_2, prime64_3,
+		prime64_4, prime32_2, prime64_5, prime32_1}
+
+	accumScalar(&xacc, p, xsecret, length)
+	// merge xacc
+	acc[1] += mix(xacc[0]^Read8(xsecret, 11), xacc[1]^Read8(xsecret, 19))
+	acc[1] += mix(xacc[2]^Read8(xsecret, 27), xacc[3]^Read8(xsecret, 35))
+	acc[1] += mix(xacc[4]^Read8(xsecret, 43), xacc[5]^Read8(xsecret, 51))
+	acc[1] += mix(xacc[6]^Read8(xsecret, 59), xacc[7]^Read8(xsecret, 67))
+
+	acc[1] = xxh3Avalanche(acc[1])
+
+	acc[0] += mix(xacc[0]^Read8(xsecret, 117), xacc[1]^Read8(xsecret, 125))
+	acc[0] += mix(xacc[2]^Read8(xsecret, 133), xacc[3]^Read8(xsecret, 141))
+	acc[0] += mix(xacc[4]^Read8(xsecret, 149), xacc[5]^Read8(xsecret, 157))
+	acc[0] += mix(xacc[6]^Read8(xsecret, 165), xacc[7]^Read8(xsecret, 173))
+	acc[0] = xxh3Avalanche(acc[0])
+
+	return acc
+}
+
+func accumScalar(xacc *[8]uint64, xinput, xsecret unsafe.Pointer, l uint64) {
+	j := uint64(0)
+
+	// Loops over block, process 16*8*8=1024 bytes of data each iteration
+	for ; j < (l-1)/1024; j++ {
+		k := xsecret
+		for i := 0; i < 16; i++ {
+			for j := uintptr(0); j < 8; j++ {
+				dataVec := Read8(xinput, 8*j)
+				keyVec := dataVec ^ Read8(k, 8*j)
+				xacc[j^1] += dataVec
+				xacc[j] += (keyVec & 0xffffffff) * (keyVec >> 32)
+			}
+			xinput, k = unsafe.Pointer(uintptr(xinput)+_stripe), unsafe.Pointer(uintptr(k)+8)
+		}
+
+		// scramble xacc
+		for j := uintptr(0); j < 8; j++ {
+			xacc[j] ^= xacc[j] >> 47
+			xacc[j] ^= Read8(xsecret, 128+8*j)
+			xacc[j] *= prime32_1
+		}
+	}
+	l -= _block * j
+
+	// last partial block (1024 bytes)
+	if l > 0 {
+		k := xsecret
+		i := uint64(0)
+		for ; i < (l-1)/_stripe; i++ {
+			for j := uintptr(0); j < 8; j++ {
+				dataVec := Read8(xinput, 8*j)
+				keyVec := dataVec ^ Read8(k, 8*j)
+				xacc[j^1] += dataVec
+				xacc[j] += (keyVec & 0xffffffff) * (keyVec >> 32)
+			}
+			xinput, k = unsafe.Pointer(uintptr(xinput)+_stripe), unsafe.Pointer(uintptr(k)+8)
+		}
+		l -= _stripe * i
+
+		// last stripe (64 bytes)
+		if l > 0 {
+			xinput = unsafe.Pointer(uintptr(xinput) - uintptr(_stripe-l))
+			k = unsafe.Pointer(uintptr(xsecret) + 121)
+
+			for j := uintptr(0); j < 8; j++ {
+				dataVec := Read8(xinput, 8*j)
+				keyVec := dataVec ^ Read8(k, 8*j)
+				xacc[j^1] += dataVec
+				xacc[j] += (keyVec & 0xffffffff) * (keyVec >> 32)
+			}
+		}
+	}
+}
+
+func mix(a, b uint64) uint64 {
+	hi, lo := bits.Mul64(a, b)
+	return hi ^ lo
+}
+func xxh3RRMXMX(h64 uint64, length uint64) uint64 {
+	h64 ^= bits.RotateLeft64(h64, 49) ^ bits.RotateLeft64(h64, 24)
+	h64 *= 0x9fb21c651e98df25
+	h64 ^= (h64 >> 35) + length
+	h64 *= 0x9fb21c651e98df25
+	h64 ^= (h64 >> 28)
+	return h64
+}
+
+func xxh64Avalanche(h64 uint64) uint64 {
+	h64 ^= h64 >> 33
+	h64 *= prime64_2
+	h64 ^= h64 >> 29
+	h64 *= prime64_3
+	h64 ^= h64 >> 32
+	return h64
+}
+
+func xxh3Avalanche(x uint64) uint64 {
+	x ^= x >> 37
+	x *= 0x165667919e3779f9
+	x ^= x >> 32
+	return x
+}
+
+func Read8(p unsafe.Pointer, offset uintptr) uint64 {
```

Comment:
better to use runtimex (this funtion is invalid in some platforms)

Reply:
fixed